robot_smach_states/src/robot_smach_states/manipulation/grab.py

```diff
@@ -278,6 +280,30 @@ def distance(e1, e2):
         return e1.distance_to_3d(e2._pose.p)
 
 
+class NavigateToPreGrasp(NavigateToDesignator):
+    """Navigate so that the arm can reach the grasp entity and we can see the entity
+    :param robot: robot object
+    :param entity_designator: designator that resolves to an Ed Entity
+    :param arm_designator: which arm to eventually grasp with?
+    """
+    def __init__(self, robot, entity_designator, arm_designator=None):
+        constraint_designator = CompoundConstraintsDesignator()
+        constraint_designator.add(ArmsreachConstraintsDesignator(robot, entity_designator, arm_designator, look=False), 'reach')
+        constraint_designator.add(LookAtConstraintsDesignator(entity_designator, 1.57), 'face') #TODO hero-specific magic number
+        super(NavigateToPreGrasp, self).__init__(robot, constraint_designator, reset_head=True)
+
+
+class NavigateToGrasp(NavigateToDesignator):
+    """Navigate so that the arm can easily grasp the entity
```

Comment:
Basically, this class replaces some other class. I think we should remove the other one, otherwise it will be unclear which one to use in the future

Reply:
The other one (navigate_to_grasp.py) is already gone